### PR TITLE
feat(nginx): Add nginx logging plugin

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
### Proposed Change
* Add a logging plugin for NGINX access + error logs. (ported from stanza-plugins)

Tested against a running NGINX instance to make sure access + error logs were parsing correctly.

As noted for Mongodb, k8s support has been removed.

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: plugins/nginx_logs.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        attributes:
+          log_type: 'nginx.access'
+        start_at: {{ .start_at }}
+        encoding: {{ .encoding }}
+        operators:
+          {{ if eq .log_format "default" }}
+          - type: regex_parser
+            regex: '^(?P<http_request_remoteIp>[^ ]*) (?P<host>[^ ]*) (?P<user>[^ ]*) \[(?P<time_local>[^\]]*)\] "(?P<http_request_requestMethod>\S+)(?: +(?P<http_request_requestUrl>[^\"]*?)(?: +(?P<http_request_protocol>\S+))?)?" (?P<http_request_status>[^ ]*) (?P<http_request_responseSize>[^ ]*)(?: "(?P<http_request_referer>[^\"]*)" "(?P<http_request_userAgent>[^\"]*)")?$'
+            timestamp:
+              parse_from: "$body.time_local"
```
Comment (cpheps) [inline plugins/nginx_logs.yaml:61]: I hit this issue with other plugins but should this be `$body.time_local`? I had to use `attributes.time_local` when the collector was using the latest logging library.

Comment (cpheps) [reply plugins/nginx_logs.yaml:61]: Same question everywhere `$body` exists.

File: plugins/nginx_logs.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                info2: 3xx
+                warn: 4xx
+                error: 5xx
+      {{end}} # end error log receiver
+
+    service:
+      pipelines:
+        logs:
+          receivers:
+          {{ if .enable_access_log }}
```
Comment (cpheps) [inline plugins/nginx_logs.yaml:124]: Does this need to be indented one?

File: plugins/nginx_logs.yaml (hunk 1 above)
Comment (BinaryFissionGames) [reply plugins/nginx_logs.yaml:61]: I think you're right, I was testing with v0.5.0, not latest.

File: plugins/nginx_logs.yaml (hunk 2 above)
Comment (BinaryFissionGames) [reply plugins/nginx_logs.yaml]: indented seems more consistent with other plugins, good catch

File: plugins/nginx_logs.yaml (hunk 1 above)
Comment (BinaryFissionGames) [reply plugins/nginx_logs.yaml]: Fixed this and tested with the latest collector; Thanks for pointing this out, didn't realise those breaking changes weren't yet made on our latest release.

**Review by cpheps — APPROVED** [review]

**Review by armstrmi — APPROVED** [review]

**Comment by BinaryFissionGames** [comment]
Had to update this PR, noticed `preserve_to` is still present in this version of the collector. In v0.50.0 of the otel collector, it should be removed, so we'll have to make some changes to the plugins to reflect that once we've upgraded our dependencies.

Here's what the logs look like now:
Access log:
```
LogRecord #99
ObservedTimestamp: 1970-01-01 00:00:00 +0000 UTC
Timestamp: 2022-05-03 16:14:02 +0000 UTC
Severity: Warn
Body: 172.17.0.2 - - [03/May/2022:12:14:02 -0400] "GET /index.html HTTP/1.1" 404 152 "-" "Go-http-client/1.1"
Attributes:
     -> http_request_userAgent: STRING(Go-http-client/1.1)
     -> http_request_requestMethod: STRING(GET)
     -> http_request_requestUrl: STRING(/index.html)
     -> http_request_remoteIp: STRING(172.17.0.2)
     -> http_request_protocol: STRING(HTTP/1.1)
     -> http_request_status: STRING(404)
     -> http_request_responseSize: STRING(152)
     -> http_request_referer: STRING(-)
     -> host: STRING(-)
     -> user: STRING(-)
     -> log_type: STRING(nginx.access)
     -> log.file.name: STRING(access.log)
Trace ID: 
Span ID: 
Flags: 0
```

Error log:
```
LogRecord #79
ObservedTimestamp: 1970-01-01 00:00:00 +0000 UTC
Timestamp: 2022-05-03 16:20:29 +0000 UTC
Severity: Info
Body: 2022/05/03 12:20:29 [info] 10549#10549: *160747 client 172.17.0.2 closed keepalive connection
Attributes:
     -> log.file.name: STRING(error.log)
     -> subrequest: STRING()
     -> pid: STRING(10549)
     -> tid: STRING(10549)
     -> referer: STRING()
     -> host: STRING()
     -> upstream: STRING()
     -> connection: STRING(160747)
     -> message: STRING(client 172.17.0.2 closed keepalive connection)
     -> server: STRING()
     -> log_type: STRING(nginx.error)
     -> client: STRING()
     -> request: STRING()
```

**Pull request merged**